ipywidgets/widgets/widget_image.py

```diff
@@ -32,3 +44,66 @@ class Image(DOMWidget, ValueWidget, CoreWidget):
     width = CUnicode(help="Width of the image in pixels.").tag(sync=True)
     height = CUnicode(help="Height of the image in pixels.").tag(sync=True)
     value = Bytes(help="The image data as a byte string.").tag(sync=True)
+
+    @classmethod
+    def from_file(cls, filename, **kwargs):
+        """
+        Create an :class:`Image` from a local file.
+
+        Parameters
+        ----------
+        filename: str
+            The location of a file to read into the value from disk.
+
+        **kwargs:
+            The keyword arguments for `Image`
+
+        Returns an `Image` with the value set from the filename.
+        """
+        value = cls._load_file_value(filename)
+
```

Comment:
What about guessing format:
```python
base, ext = os.path.splitext(filename)
if ext and 'format' not in kwargs:
    kwargs = dict(**kwargs, format=ext[1:])  # copy dict and remove leading .
```

Reply:
@maartenbreddels  Good idea, though I think the implementation needs to be a bit more complicated - it should try to map the extension to [the MIME type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types), which is not always the same thing.

I'll fix that up.

Reply:
Yes, we can choose maybe from a whitelist (all the ones starting with image/). Maybe give a warning if it's not known?

Reply:
You could just use the python library for mapping filenames to formats: https://docs.python.org/3/library/mimetypes.html#mimetypes.guess_type